Ros2 devel

rjshim opened the pull request:
I added the following.

1. turtlebot3_manipulation_controller (controller)
2. turtlebot3_manipulation_description (rviz) 
3. turtlebot3_manipulation_teleop (teleop)
4. naming change (open_manipulator_with_tb3 -> turtlebot3_manipulation)

Thank you in advance for reviewing the source code!
Ryan

robotpilot commented on turtlebot3_manipulation/CHANGELOG.rst:
```diff
@@ -1,7 +1,11 @@
+2.0.1 (2019-12-31)
+------------------
+* 
+
```
Don't put in future plans.

rjshim replied:
Removed

robotpilot commented on turtlebot3_manipulation/CHANGELOG.rst:
```diff
@@ -1,7 +1,11 @@
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Changelog for package open_manipulator_with_tb3
+Changelog for package turtlebot3_manipulation
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
The '^' mark should match the letter size.

rjshim replied:
Fixed

robotpilot commented on turtlebot3_manipulation/package.xml:
```diff
@@ -0,0 +1,24 @@
+  <name>turtlebot3_manipulation</name>
+  <version>2.0.1</version>
+  <description>
+    ROS2 package for turtlebot3_manipulation
```
```suggestion
    ROS 2 package for turtlebot3_manipulation
```
Can you write it in more detail? This can be found by users by searching.

rjshim replied:
Fixed

robotpilot commented on turtlebot3_manipulation/package.xml:
```diff
@@ -0,0 +1,24 @@
+  </description>
+  <license>Apache 2.0</license>
+  <author email="thlim@robotis.com">Ryan Shim</author>
+  <author email="thlim@robotis.com">Darby Lim</author>
```
Make sure the new author doesn't come first.

rjshim replied:
Fixed

robotpilot commented on turtlebot3_manipulation_controller/CHANGELOG.rst:
```diff
@@ -1,7 +1,11 @@
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Changelog for package open_manipulator_with_tb3_tools
+Changelog for package turtlebot3_manipulation_controller
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
A shorter '^' than the above string is a syntax error.

rjshim replied:
Changed

robotpilot commented on turtlebot3_manipulation_controller/CHANGELOG.rst:
```diff
@@ -1,7 +1,11 @@
+2.0.1 (2019-12-31)
+------------------
+* ???
+
```
Delete it.

rjshim replied:
Removed

robotpilot commented on turtlebot3_manipulation_controller/launch/turtlebot3_manipulation_controller.launch.py:
```diff
@@ -0,0 +1,61 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Author: Ryan Shim
```
Why is OSRF here? What is it related to?

rjshim replied:
Changed

robotpilot commented on turtlebot3_manipulation_controller/package.xml:
```diff
@@ -1,21 +1,21 @@
+    ROS 2 package for turtlebot3_manipulation_controller
   </description>
   <license>Apache 2.0</license>
+  <author email="jhshim@robotis.com">Ryan Shim</author>
```
Please move to the next line.

rjshim replied:
Changed

robotpilot commented on turtlebot3_manipulation_description/launch/turtlebot3_manipulation_rviz.launch.py:
```diff
@@ -0,0 +1,59 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Author: Ryan Shim
```
Same as above

rjshim replied:
Changed

robotpilot commented on turtlebot3_manipulation_description/package.xml:
```diff
@@ -0,0 +1,25 @@
+    ROS2 package for turtlebot3_manipulation_description
+  </description>
+  <license>Apache 2.0</license>
+  <author email="jhshim@robotis.com">Ryan Shim</author>
```
Same as above

rjshim replied:
Changed

robotpilot commented on turtlebot3_manipulation_description/CHANGELOG.rst:
```diff
@@ -2,6 +2,10 @@
 Changelog for package open_manipulator_with_tb3_description
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+2.0.1 (2019-12-31)
```
Please delete it.

rjshim replied:
Removed

robotpilot commented on turtlebot3_manipulation_teleop/launch/turtlebot3_manipulation_teleop_joystick.launch.py:
```diff
@@ -0,0 +1,37 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Author: Ryan Shim
```
Change your license.

rjshim replied:
Changed

robotpilot commented on turtlebot3_manipulation_teleop/launch/turtlebot3_manipulation_teleop_keyboard.launch.py:
```diff
@@ -0,0 +1,31 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Author: Ryan Shim
```
You write OSRF in all launch files, which is incorrect. 
Please change it. I will not describe it as a duplicate of another file.

rjshim replied:
Changed

robotpilot commented on turtlebot3_manipulation_teleop/package.xml:
```diff
@@ -0,0 +1,25 @@
+  <url type="website">http://wiki.ros.org/open_manipulator_libs</url>
+  <url type="emanual">http://emanual.robotis.com/docs/en/platform/openmanipulator</url>
+  <url type="repository">https://github.com/ROBOTIS-GIT/open_manipulator</url>
+  <url type="bugtracker">https://github.com/ROBOTIS-GIT/open_manipulator/issues</url>
```
Why is Open Manipulator here?

rjshim replied:
Changed

robotpilot commented:
@rjshim  
Ryan! Thank you very much. This repository name and package have been renamed differently. I understood why it changed. How about creating a new repository? What do you think?

rjshim commented:
@robotpilot @routiful 
I did the following. 
1. Modified changelog files
2. Modified package.xml files
3. Modified licenses in launch files

I took @routiful's advice and didn't make a new repository for tb3_manipulation ROS 2 packages as it will be easier to maintain them.

Ryan :)
